src/transformers/models/speecht5/modeling_speecht5.py

```diff
@@ -695,9 +690,7 @@ def forward(
 
         if speaker_embeddings is not None:
             speaker_embeddings = nn.functional.normalize(speaker_embeddings)
-            speaker_embeddings = speaker_embeddings.unsqueeze(1)
-            speaker_embeddings = speaker_embeddings.expand(-1, inputs_embeds.size(1), -1)
-            speaker_embeddings = speaker_embeddings.repeat(inputs_embeds.size(0), 1, 1)
```

Comment:
The way I see it, there are two possible situations:
1. As many speaker embeddings as sample (i.e `batch size = number of speaker embeddings`)
2. One-to-many speaker embeddings (i.e one speaker embeddings for every sample of the batch)

Your proposed solution addresses situation 1., and the previous solution addressed situation 2.

To be complete, we should have a code that addresses both situations ! (and that throws an `Error` in other cases).